### PR TITLE
Feature/minimal app

### PR DESCRIPTION
**Pull request by FrackiewiczP**
I've done some renaming along the way because naming classes `Task` and `User` causes lots of name clashes.

### CODE REVIEW

File: docs/guides/ExampleApp/backend/release/Dockerfile.api [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM mcr.microsoft.com/dotnet/aspnet:6.0
```
Comment (Saancreed) [inline docs/guides/ExampleApp/backend/release/Dockerfile.api:1]: Shouldn't we prefer deployments via Argo nowadays? That would make this entire directory unnecessary, just like "deploy to test" step of the Jenkinsfile.

But if we don't, then this should probably use .NET 7.0 just like dev env images.

**Review by Saancreed — COMMENTED** [review]
Terraform state files and provider binaries should probably be removed from git and added to gitignore.

File: docs/guides/ExampleApp/backend/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    copyArtifactPermission('/ExampleApp/Production/*')
+]);
+
+def isReleaseBuild = BRANCH_NAME == 'master';
+def dbPassword = 'Passw12#'
+
+leancode.builder('exampleapp-backend')
+    .withMonorepoTest(['backend'])
+    .withBuildKit()
+    .withDotnet([version: '6.0', image: 'mcr.microsoft.com/dotnet/sdk'])
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/Jenkinsfile:11]: 7.0?

File: docs/guides/ExampleApp/backend/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# ExampleApp
+
+>
+> TODO: put here some meaningful quote
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/README.md:4]: You could do this one :see_no_evil:

File: docs/guides/ExampleApp/backend/dev/Dockerfile.api [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg
+
+RUN groupadd --gid 1000 dotnet \
+    && useradd -s /bin/bash --uid 1000 --gid 1000 -m dotnet
+RUN mkdir /app && chown dotnet:dotnet /app
+USER dotnet
+
+ENV PATH="$PATH:/home/dotnet/.dotnet/tools"
+RUN dotnet tool install -g dotnet-ef --version '6.0.0-*'
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/dev/Dockerfile.api:14]:
```suggestion
RUN dotnet tool install -g dotnet-ef --version '7.0.0-*'
```

File: docs/guides/ExampleApp/backend/dev/Dockerfile.integration_tests [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg
+
+RUN groupadd --gid 1000 dotnet \
+    && useradd -s /bin/bash --uid 1000 --gid 1000 -m dotnet
+RUN mkdir /app && chown dotnet:dotnet /app
+USER dotnet
+
+ENV PATH="$PATH:/home/dotnet/.dotnet/tools"
+RUN dotnet tool install -g dotnet-ef --version '6.0.0-*'
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/dev/Dockerfile.integration_tests:14]:
```suggestion
RUN dotnet tool install -g dotnet-ef --version '7.0.0-*'
```

File: docs/guides/ExampleApp/backend/dev/Dockerfile.migrations [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg
+
+RUN groupadd --gid 1000 dotnet \
+    && useradd -s /bin/bash --uid 1000 --gid 1000 -m dotnet
+RUN mkdir /app && chown dotnet:dotnet /app
+USER dotnet
+
+ENV PATH="$PATH:/home/dotnet/.dotnet/tools"
+RUN dotnet tool install -g dotnet-ef --version '6.0.0-*'
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/dev/Dockerfile.migrations:14]:
```suggestion
RUN dotnet tool install -g dotnet-ef --version '7.0.0-*'
```

File: docs/guides/ExampleApp/backend/dev/Dockerfile.tests [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+RUN curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg
+
+RUN groupadd --gid 1000 dotnet \
+    && useradd -s /bin/bash --uid 1000 --gid 1000 -m dotnet
+RUN mkdir /app && chown dotnet:dotnet /app
+
+USER dotnet
+
+ENV PATH="$PATH:/home/dotnet/.dotnet/tools"
+RUN dotnet tool install -g dotnet-ef --version '6.0.0-*'
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/dev/Dockerfile.tests:15]: unnecessary here?
```suggestion
RUN dotnet tool install -g dotnet-ef --version '7.0.0-*'
```

File: docs/guides/ExampleApp/backend/dev/config/integration_tests.sh [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+export Domains__Public=localhost
+export Domains__ApiInternal=localhost
+
+#TODO: Fix once corelib is fixed
+export SqlServer__ConnectionStringBase='Server=mssql-svc.shared.svc.cluster.local,1433;User Id=sa;Password=Passw12#;Encrypt=false'
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/dev/config/integration_tests.sh:7]: Does this TODO still stands?

File: docs/guides/ExampleApp/backend/src/Core/ExampleApp.Core.Services/DataAccess/CoreDbContext.cs [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+                    inner.Property(a => a.Name).HasMaxLength(500);
+                    inner.Property(a => a.Id).IsTypedId();
+                    inner.Property(a => a.ParentProjectId).IsTypedId();
+                    inner.Property(a => a.AssignedEmployeeId).IsTypedId();
+
+                    inner.HasKey(a => new { a.Id }).IsClustered(false);
+                    inner.HasIndex(a => new { a.ParentProjectId, a.Id }).IsClustered(true);
+
+                    inner.ToTable("Assignments");
```
Comment (lukaszgarstecki) [inline docs/guides/ExampleApp/backend/src/Core/ExampleApp.Core.Services/DataAccess/CoreDbContext.cs:67]: NIT: `Project_Assignments`?

**Review by lukaszgarstecki — COMMENTED** [review]
I guess we want to propagate small adjustments made here into corelibrary-template. Do we want to treat this minimal app just the same like other projects and address the changes both here and in template repo @jakubfijalkowski ? Or do we plan some other approach to template and we want to skip adjustments now?

**Comment by jakubfijalkowski** [comment]
Superseded by dedicated repository with an example.

**Pull request closed without merging**